bench/GithubApiBenchmark.cs

```diff
@@ -115,26 +116,102 @@ public class GithubApiBenchmark
+                              new Author(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15, p16));
+
+    static readonly IJsonReader<Stats> StatsJsonReader =
+        JsonReader.Object(JsonReader.Property("additions", JsonReader.Int32()),
+                          JsonReader.Property("deletions", JsonReader.Int32()),
+                          JsonReader.Property("total", JsonReader.Int32()),
+                          (p1, p2, p3) => new Stats(p1, p2, p3));
+
+    static readonly IJsonReader<File> FileJsonReader =
+        JsonReader.Object(JsonReader.Property("filename", JsonReader.String()),
+                          JsonReader.Property("additions", JsonReader.Int32()),
+                          JsonReader.Property("deletions", JsonReader.Int32()),
+                          JsonReader.Property("changes", JsonReader.Int32()),
+                          JsonReader.Property("status", JsonReader.String()),
+                          JsonReader.Property("raw_url", UriReader),
+                          JsonReader.Property("blob_url", UriReader),
+                          JsonReader.Property("patch", JsonReader.String()),
+                          (p1, p2, p3, p4, p5, p6, p7, p8) =>
+                              new File(p1, p2, p3, p4, p5, p6, p7, p8));
 
     static readonly IJsonReader<MergeBranchResponse> MergeBranchResponseJsonReader =
-        JsonReader.Object(JsonReader.Property("url", from s in JsonReader.String() select new Uri(s)),
+        JsonReader.Object(JsonReader.Property("url", UriReader),
                           JsonReader.Property("sha", JsonReader.String()),
+                          JsonReader.Property("node_id", JsonReader.String()),
+                          JsonReader.Property("html_url", UriReader),
+                          JsonReader.Property("comments_url", UriReader),
                           JsonReader.Property("commit", CommitJsonReader),
-                          (url, sha, commit) => new MergeBranchResponse(url, sha, commit));
+                          JsonReader.Property("author", AuthorJsonReader),
+                          JsonReader.Property("committer", AuthorJsonReader),
+                          JsonReader.Property("parents", ImmutableArrayReader(TreeJsonReader)),
+                          JsonReader.Property("stats", StatsJsonReader),
+                          JsonReader.Property("files", ImmutableArrayReader(FileJsonReader)),
+                          (p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11) =>
+                              new MergeBranchResponse(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11));
+
+    static IJsonReader<T?> Nullable<T>(IJsonReader<T> reader, T? @null) =>
+        JsonReader.Either(JsonReader.Null(@null), from v in reader select v);
+
+    static IJsonReader<ImmutableArray<T>> ImmutableArrayReader<T>(IJsonReader<T> reader) =>
+        from v in JsonReader.Array(reader) select v.ToImmutableArray();
```

Comment:
This overload will avoid going from a list to an array to an immutable array by skipping over the intermediate list to array conversion:

```suggestion
        JsonReader.Array(reader, list => list.ToImmutableArray());
```